### PR TITLE
feature: list only Zaak UUIDs for reindexing

### PR DESCRIPTION
**Pull request by hsiliev**
Ignore exceptions during object conversion

Solves: PZ-3493

### CODE REVIEW

**Review by edgarvonk — CHANGES_REQUESTED** [review]

File: src/main/java/net/atos/client/zgw/zrc/model/ZaakUuid.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+import java.util.UUID;
+
+/**
+ * Zaak UUID
+ *
+ * @param uuid Unieke resource identifier (UUID4)
+ */
+public record ZaakUuid(UUID uuid) {
```
[the remark this answers is not in the document]
Comment (hsiliev) [reply src/main/java/net/atos/client/zgw/zrc/model/ZaakUuid.java:16]: Using just UUID is probably possible, but requires a custom deserializer as far as I can tell. Don't think it is worth the effort.

Instead we use the fact that this object can use the listZaken payload but use deserialization only for UUID

File: src/main/kotlin/net/atos/zac/zoeken/IndexingService.kt [hunk 2]
```diff
@@ -88,7 +91,10 @@ class IndexingService @Inject constructor(
      * @param performCommit whether to perform a hard Solr commit
      */
     fun indexeerDirect(objectIds: List<String>, objectType: ZoekObjectType, performCommit: Boolean) =
-        addToSolrIndex(objectIds.map { getConverter(objectType).convert(it) }, performCommit)
+        addToSolrIndex(
+            objectIds.map { continueOnExceptions(objectType) { getConverter(objectType).convert(it) } },
```
[the remark this answers is not in the document]
Comment (hsiliev) [reply src/main/kotlin/net/atos/zac/zoeken/IndexingService.kt:95]: We use map to loop over all IDs and load all the objects.
Refactored this part a bit to make this more obvious.

File: src/main/kotlin/net/atos/zac/zoeken/IndexingService.kt [hunk 3]
```diff
@@ -228,25 +234,25 @@ class IndexingService @Inject constructor(
             ZGWApiService.FIRST_PAGE_NUMBER_ZGW_APIS
 
         for (pageNumber in ZGWApiService.FIRST_PAGE_NUMBER_ZGW_APIS..numberOfPages) {
-            continueOnExceptions(ZoekObjectType.ZAAK) { reindexZakenPage(pageNumber, numberOfPages) }
+            continueOnExceptions(ZoekObjectType.ZAAK) { reindexZakenPage(pageNumber, numberOfZaken) }
         }
     }
 
     private fun reindexZakenPage(pageNumber: Int, totalSize: Int) {
```
[the remark this answers is not in the document]
Comment (hsiliev) [reply src/main/kotlin/net/atos/zac/zoeken/IndexingService.kt]: Changed to totalCount

File: src/main/kotlin/net/atos/zac/zoeken/IndexingService.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (edgarvonk) [reply src/main/kotlin/net/atos/zac/zoeken/IndexingService.kt]: clear!

**Review by edgarvonk — APPROVED** [review]
very nice

**Pull request merged**